src/utils/unist.mjs

```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+import { pointEnd, pointStart } from 'unist-util-position';
+
+/**
+ * This utility allows us to join children Nodes into one
+ * and transfor them back to what their source would look like
+ *
+ * @param {Array<import('unist').Parent & import('unist').Literal>} nodes Nodes to parsed and joined
+ * @returns {string} The parsed and joined nodes as a string
+ */
+export const transformNodesToString = nodes => {
+  const mappedChildren = nodes.map(node => {
+    if (node.type === 'inlineCode') {
+      return `\`${node.value}\``;
+    }
+
+    if (node.type === 'strong') {
+      return `**${transformNodesToString(node.children)}**`;
+    }
+
+    if (node.type === 'emphasis') {
+      return `_${transformNodesToString(node.children)}_`;
+    }
+
+    if (node.children) {
+      return transformNodesToString(node.children);
+    }
+
+    return node.value;
```

Comment:
```suggestion
    switch (node.type) {
      case 'inlineCode':
        return `\`${node.value}\``;
      case 'strong':
        return `**${transformNodesToString(node.children)}**`;
      case 'emphasis':
        return `_${transformNodesToString(node.children)}_`;
      default:
        return node.children
          ? transformNodesToString(node.children)
          : node.value;
    }
```

We can use switch case instead of multiple if conditions